assets/js/modules/analytics/datastore/properties.js

```diff
@@ -0,0 +1,325 @@
+/**
+ * modules/analytics data store: properties.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './index';
+
+// Actions
+const FETCH_CREATE_PROPERTY = 'FETCH_CREATE_PROPERTY';
+const FETCH_PROPERTIES_PROFILES = 'FETCH_PROPERTIES_PROFILES';
+const RECEIVE_CREATE_PROPERTY = 'RECEIVE_CREATE_PROPERTY';
+const RECEIVE_CREATE_PROPERTY_FAILED = 'RECEIVE_CREATE_PROPERTY_FAILED';
+const RECEIVE_PROPERTIES_PROFILES = 'RECEIVE_PROPERTIES_PROFILES';
+const RECEIVE_PROPERTIES_PROFILES_FAILED = 'RECEIVE_PROPERTIES_PROFILES_FAILED';
+
+export const INITIAL_STATE = {
+	isDoingCreateProperty: {},
+	isFetchingPropertiesProfiles: {},
+	properties: undefined,
+};
+
+export const actions = {
+	/**
+	 * Creates a new Analytics property.
+	 *
+	 * Creates a new Analytics property for an existing Google Analytics account.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} accountId Google Analytics account ID.
+	 * @return {Object} Redux-style action.
+	 */
+	*createProperty( accountId ) {
```

Comment:
Note that all parameters should be named according to _our_ naming conventions unless they come from an API response via Google. In other words, this should be `accountID` throughout, except for when looking at API responses where this is actually `accountId`. A bit annoying yes, but we need to stick to our conventions.